File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionTransform.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionTransform extends TransformerFunction {
    public TransformerFunctionTransform() {
        super(FunctionDescription.of(
            Map.of(
            "to", ArgumentType.of(ArgType.Transformer).position(0).defaultIsNull(true)
            )
        ));
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        var input = context.getJsonElement(null);
        var to = context.getJsonElement("to", false);
        return context.transformItem(to, input);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionTransform.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { ArgType } from "./common/ArgType";

class TransformerFunctionTransform extends TransformerFunction {
  constructor() {
    super({
      arguments: {
        to: { type: ArgType.Transformer, position: 0, defaultIsNull: true },
      },
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const input = await context.getJsonElement(null);
    const to = await context.getJsonElement("to", false);
    return context.transformItem(to, input);
  }
}

export default TransformerFunctionTransform;
